tests/metadata/comparator.py

```diff
@@ -0,0 +1,66 @@
+from pyshelf.metadata.yaml_codec import YamlCodec
+from pyshelf.metadata.mapper import Mapper
+from pyshelf.metadata.cloud_portal import CloudPortal
+from pyshelf.resource_identity import ResourceIdentity
+from urlparse import urlparse
+from pyshelf.search.metadata import Metadata
+from elasticsearch import Elasticsearch
+from pyshelf.cloud.storage import Storage
+
+
+class Comparator(object):
+    def __init__(self, test, es_connection_string, logger):
+        self._es_connection_parts = urlparse(es_connection_string)
+        self.logger = logger
+        self.test = test
+        self._fake_container = None
+        self._es_connection = None
+        self._cloud_portal = None
+
+    @property
+    def fake_container(self):
+        if not self._fake_container:
+            # Trailing comma in the tuple is important otherwise it is interpretted
+            # as a grouping and just returns the type "object"
+            self._fake_container = type("FakeMetadataContainer", (object,), {})()
+            self._fake_container.yaml_codec = YamlCodec()
+            self._fake_container.mapper = Mapper()
+            self._fake_container.create_cloud_storage = lambda: Storage(None, None, "test", self.logger)
```

Comment:
Sweet. `tests.routes.artifact_test.ArtifactTest.test_artifact_upload_other_bucket` tests uploading to another bucket. No metadata tests but yea would be weird. Thanks gonna merge this up.